usecases/auth/authorization/rbac/authorizer.go

```diff
@@ -77,3 +90,81 @@ func (m *manager) Authorize(principal *models.Principal, verb string, resources
 func (m *manager) AuthorizeSilent(principal *models.Principal, verb string, resources ...string) error {
 	return m.authorize(principal, verb, true, resources...)
 }
+
+func (m *manager) FilterAuthorizedResources(principal *models.Principal, verb string, resources ...string) ([]string, error) {
+	if principal == nil {
+		return nil, errors.NewUnauthenticated()
+	}
+
+	if len(resources) == 0 {
+		return nil, fmt.Errorf("at least 1 resource is required")
+	}
+
+	logger := m.logger.WithFields(logrus.Fields{
+		"action":         "authorize",
+		"user":           principal.Username,
+		"component":      authorization.ComponentName,
+		"request_action": verb,
+	})
+	if len(principal.Groups) > 0 {
+		logger = logger.WithField("groups", principal.Groups)
+	}
+
+	var permResults []logrus.Fields
+	var allowedResources []string
+
+	// First try with wildcard
+	parts := strings.Split(resources[0], "/")
+	parts[len(parts)-1] = "*"
+	wildcardPath := strings.Join(parts, "/")
```

Comment:
I don't think this is valid in every case. Eg if we have resources like this:

class A, Tenant B
class C, Tenant D

With the current code we would check if the caller has permissions for Class A, tenant * and if yes also allow access to class C, Tenant D no matter what permissions we have for that. 

Proposal: 
We do
```
func (m *manager) FilterAuthorizedResources(principal *models.Principal, verb string, wildcardPath string, resources ...string) ([]string, error) {
```
where the caller defines `wildcardPath`. Then, the caller has full control and can provide this path if it makes sense and not otherwise

Reply:
But this still has the same problem as before:

Two resources are requestes:
- classA/tenants/TenantB
- classC/tenants/TenantD

the wildcard check will try `classA/tenants/*` and if that succeeds allow access to `classC/tenants/TenantD` even though the user might not have any rights for classC at all.

Reply:
it won't because the original path comes to the request shall include the needed checks, so 

request comes for `schema/class/ClassA/tenants/tenant1` and we convert it here to `schema/class/ClassA/tenants/*` this will check if the user has access to tenants inside the `classA` only and it won't succeed if the user has access to `classB`

there is actually an acceptance test `TestAuthzRoleFilteredTenantPermissions` does that [here](https://github.com/weaviate/weaviate/pull/7196/files#diff-5fe4ee7308ebe082ea10c6b45c5864857752d51d3417a7a0a5acb9ec4b351e5dR919)

Reply:
`FilterAuthorizedResources()` is not made to verify resources from different entities, it's to filter from the same entity, what we can do we return error if there is a situation like this for now given we don't need to use this in different areas

Reply:
this feels like a footgun and would be very unexpected. Why not offer the caller to simply provide the path with the wildcard? The caller knows what is needed and when it makes sense or not. And it is very simple to implement

Reply:
> this feels like a footgun and would be very unexpected. 

again this method is not made to be called by different resources, this is just used for `Filter()` so even to protect users of this method we can simply return error. if we found down the road that we need to globalize it we can introduce the called with the path. 

Why not introducing it now, because it won't add any value!
there is one of 2 options 
- either there is wildcard path which means all of the resources belong to the same entity
- or there is no wildcard path  which means there is no need to check the check in 1st place

Reply:
> Why not introducing it now, because it won't add any value!

a) because people reuse things and wont know this when working on something related in a year.
b) It seems much simpler than `WildcardPath` which would be another place that needs to be adapted with new paths/domains

I don't see any upsides of creating the wildcard paths here but downsides

Reply:
I took this bit totally out of the FilterAuthZ function and moved to Filter instead